src/main/java/io/cdap/plugin/gcp/firestore/common/FirestoreConfig.java

```diff
@@ -37,27 +38,39 @@ public class FirestoreConfig extends PluginConfig {
   @Nullable
   protected String project;
 
+  @Name(NAME_DATABASE)
+  @Description("Name of the Firestore Database. "
+    + "If not specified, it will use '(default)'.")
+  @Macro
+  protected String databaseName;
+
   @Name(NAME_SERVICE_ACCOUNT_TYPE)
   @Description("Service account type, file path where the service account is located or the JSON content of the " +
     "service account.")
-  @Macro
-  @Nullable
   protected String serviceAccountType;
 
   @Name(NAME_SERVICE_ACCOUNT_FILE_PATH)
   @Description("Path on the local file system of the service account key used "
     + "for authorization. Can be set to 'auto-detect' when running on a Dataproc cluster. "
     + "When running on other clusters, the file must be present on every node in the cluster.")
-  @Macro
   @Nullable
   protected String serviceFilePath;
 
   @Name(NAME_SERVICE_ACCOUNT_JSON)
   @Description("Content of the service account file.")
-  @Macro
   @Nullable
   protected String serviceAccountJson;
 
+  @Name(SERVICE_ACCOUNT_FILE_PATH)
+  @Description("File path of the Service Account JSON file.")
+  @Nullable
+  protected String serviceAccountFilePath;
```

Comment:
The `serviceFilePath` field is meant do the same, please remove duplicate field.